cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleServiceTest.java

```diff
@@ -452,6 +473,102 @@ public void testCreateAppDetailsArchive() throws Exception {
     deleteNamespace("ns3");
   }
 
+  @Test(expected = ApplicationNotFoundException.class)
+  public void testDeployWithoutMarkingAppAsLatest() throws Exception {
+    String appName = "application_not_marked_latest";
+    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, appName);
+    Id.Artifact artifactId = Id.Artifact.from(Id.Namespace.DEFAULT, "appWithConfig", "1.0.0-SNAPSHOT");
+    addAppArtifact(artifactId, ConfigTestApp.class);
+    HttpResponse resp = deployWithoutMarkingLatest(
+        appId, new AppRequest<>(ArtifactSummary.from(artifactId.toArtifactId())));
+    Assert.assertEquals(200, resp.getResponseCode());
+
```

Comment:
Can you please put a comment here why `ApplicationNotFoundException` is expected here. Usually happy path tests should not have exception check so having a comment would be useful for future contributors. 

Also this tests a new app I believe? We also need to check adding versions for a existing app

Reply:
We should also verify the version was actually created with latest field as false

Reply:
As discussed offline, I have added comments describing the intended behaviour of the test.